torch/distributions/distribution.py

```diff
@@ -273,8 +273,15 @@ def _validate_sample(self, value):
                           '`validate_args=False` to turn off validation.')
             return
         assert support is not None
-        if not support.check(value).all():
-            raise ValueError('The value argument must be within the support')
+        valid = support.check(value)
+        if not valid.all():
+            invalid_value = value.data[~valid]
+            error = ValueError(
```

Comment:
Do you think we should also specify the support in the error message for arguments checking in `__init__`?

Reply:
done